test/functional/rpc_blockchain.py

```diff
@@ -549,6 +549,7 @@ def _test_waitforblock(self):
         # The chain has probably already been restored by the time reconsiderblock returns,
         # but poll anyway.
         self.wait_until(lambda: node.waitfornewblock(timeout=100)['hash'] == current_hash)
+        assert_raises_rpc_error(-1, "Negative timeout", lambda: node.waitfornewblock(timeout=-1))
```

Comment:
Why do we need the `lambda` here? It will throw the error right at the beginning of the execution.

Reply:
Could try:

```py
assert_raises_rpc_error(-1, "Negative timeout", node.waitfornewblock, -1)
```

Reply:
thanks! that works updated in [93747d9](https://github.com/bitcoin/bitcoin/pull/31746/commits/93747d934b8a5a1732732a958c0d7e2f5dd0b8c2)